### PR TITLE
[GR-44098] Add lambda methods test in debuginfotest

### PR DESCRIPTION
**Pull request by zakkak**
Catches cases like https://github.com/graalvm/mandrel/issues/355 caused by [JDK-8281266](https://bugs.openjdk.java.net/browse/JDK-8281266)

### CODE REVIEW

**Comment by zakkak** [comment]
The underlying issue has been [fixed in jdk17u-dev](https://github.com/openjdk/jdk17u-dev/pull/274) and is expected to land in OpenJDK 17.0.4 (July 19 2022). As a result we could consider this PR for inclusion in GraalVM 22.2.0 (July 26, 2022).

**Comment by oubidar-Abderrahim** [comment]
Hi, Thank you for contributing to GraalVM, when this PR is ready for review, please let me know so that I can take care of it and assign it to the proper engineer.

**Comment by olpaw** [comment]
Hi @zakkak,
while already at adding a test for lambdas it would be great to also test setting breakpoints in lambdas via

- a line breakpoint in the body of the lambda function
- method breakpoint on the name of the lambda function

File: substratevm/src/com.oracle.svm.test/src/hello/Hello.java
Comment (olpaw) [inline substratevm/src/com.oracle.svm.test/src/hello/Hello.java:233]: I do not see any use of that lambda elsewhere. Thus, it is not guaranteed that the reachability analysis will keep it.

**Review by adinn — COMMENTED** [review]
The current check is good. However, as Paul said, we really need this to check breakpoints work at the point of lambda invocation and in the lambda itself. I think it would be best to add that from the start rather than via a follow-up.

File: substratevm/src/com.oracle.svm.test/src/hello/Hello.java
Comment (adinn) [reply substratevm/src/com.oracle.svm.test/src/hello/Hello.java:233]: Yes, this actually needs to be called by the test program.

**Comment by zakkak** [comment]
Hi all, this is on my to-do list (just not high enough :) )

File: substratevm/mx.substratevm/testhello.py
Comment (olpaw) [inline substratevm/mx.substratevm/testhello.py:128]: To avoid all those line-number change related changes we should use anchors.

Define a python function like
```python
def lineNumberOf(filename, regex):
    ....
    return line_number_where_given_regex_matches_or_error_if_not_found_or_found_more_than_once
```

Comment (olpaw) [reply substratevm/mx.substratevm/testhello.py:128]: With something like that we could prevent the need for changes just because some part of `Hello.java` has new lines somewhere in the beginning.

Comment (zakkak) [reply substratevm/mx.substratevm/testhello.py:128]: We already do something like this for checking listings and backtraces, by not specifying the line number and expecting a pattern that matches the line we want, e.g.: https://github.com/oracle/graal/blob/3e7732200f48d72a434e69b0d89ffa997e7d23fb/substratevm/mx.substratevm/testhello.py#L214

As a result this suggestion would only make sense for setting breakpoints at specific lines like we do in: https://github.com/oracle/graal/blob/3e7732200f48d72a434e69b0d89ffa997e7d23fb/substratevm/mx.substratevm/testhello.py#L605 

https://github.com/oracle/graal/blob/3e7732200f48d72a434e69b0d89ffa997e7d23fb/substratevm/mx.substratevm/testhello.py#L630

and

https://github.com/oracle/graal/blob/3e7732200f48d72a434e69b0d89ffa997e7d23fb/substratevm/mx.substratevm/testhello.py#L661 

The reason we explicitly set the constants is to be more verbatim on what we expect and be able to catch glitches that might otherwise get away (e.g. line numbers being wrong by one).

Comment (zakkak) [reply substratevm/mx.substratevm/testhello.py:128]: @olpaw WDYT about ^^

Comment (olpaw) [reply substratevm/mx.substratevm/testhello.py:128]: > The reason we explicitly set the constants is to be more verbatim on what we expect and be able to catch glitches that might otherwise get away (e.g. line numbers being wrong by one).

Alright. I leave it up to you if you want to keep explicit line numbers here.

**Review by olpaw — APPROVED** [review]

**Comment by olpaw** [comment]
@zakkak please fix the merge conflict before we make the internal PR.

**Comment by zakkak** [comment]
> @zakkak please fix the merge conflict before we make the internal PR.

Done, should be good to go.

**Review by adinn — APPROVED** [review]

**Comment by zakkak** [comment]
@olpaw could you please create the internal PR for this?

**Comment by zakkak** [comment]
@olpaw @oubidar-Abderrahim I rebased once again to resolve some new conflicts. Please review and merge if OK.

**Comment by zakkak** [comment]
Ping @olpaw @oubidar-Abderrahim ^^

**Comment by oubidar-Abderrahim** [comment]
Hi @zakkak , it seems there are some more conflicts to solve, please do so when possible.
@olpaw Once the conflicts are solved, can we integrate this PR into main, or are there any more changes required?

**Comment by zakkak** [comment]
> Hi @zakkak , it seems there are some more conflicts to solve, please do so when possible.

@oubidar-Abderrahim done. Please note that this is the 3rd rebase after the PR's approval. Since this has already been approved, and the conflicts were mostly on line number differences, could you please push it for internal review once the github actions run?

**Comment by zakkak** [comment]
@oubidar-Abderrahim the CI is happy and there are no conflicts, could you please push this to the internal queue?

**Comment by oubidar-Abderrahim** [comment]
Hi @zakkak, there are some gate failures that're blocking this PR's merge
<details>
<summary>gate-svm-enterprise-debuginfotest-jdk20-linux-amd64</summary>

```
Running: mx --kill-with-sigquit --strict-compliance --dy substratevm-enterprise-gcs debuginfotest --output-path /b/b/e/main/substratevm-enterprise/svmbuild -H:+UseSerialGC -H:+UseCompressedReferences
path=/b/b/e/main/substratevm-enterprise/svmbuild
sourcepath=/b/b/e/graal/substratevm/src/com.oracle.svm.test/src
sourcecache=/b/b/e/main/substratevm-enterprise/svmbuild/sources
native_image ['--native-image-info', '-H:Path=/b/b/e/main/substratevm-enterprise/svmbuild', '-H:+VerifyNamingConventions', '-cp', '/sc/bb/mx/HAMCREST_e237ae735aac4fa5a7253ec693191f42ef7ddce384c11d29fbf605981c0be077d086757409acad53cb5b9e53d86a07cc428d459ff0f5b00d32a8cbbca390be49/hamcrest.jar:/sc/bb/mx/JUNIT_5974670c3d178a12da5929ba5dd9b4f5ff461bdc1b92618c2c36d53e88650df7adbf3c1684017bb082b477cb8f40f15dcf7526f06f06183f93118ba9ebeaccce/junit.jar:/b/b/e/mxtool/mxbuild/jdk20/dists/jdk9/junit-tool.jar:/b/b/e/graal/substratevm/mxbuild/jdk20/com.oracle.svm.test/bin', '-Dgraal.LogFile=graal.log', '-g', '-H:-OmitInlinedMethodDebugLineInfo', '-H:+SourceLevelDebug', '-H:DebugInfoSourceSearchPath=/b/b/e/graal/substratevm/src/com.oracle.svm.test/src', '-H:DebugInfoSourceCacheRoot=/b/b/e/main/substratevm-enterprise/svmbuild/sources', 'hello.Hello', '-H:+UseSerialGC', '-H:+UseCompressedReferences', '-Dllvm.bin.dir=/b/b/e/graal/sdk/mxbuild/linux-amd64/LLVM_TOOLCHAIN/bin/', '-H:+SpawnIsolates']
========================================================================================================================
GraalVM Native Image: Generating 'hello.hello' (executable)...
========================================================================================================================
[1/8] Initializing...                                                                                   (14.3s @ 0.24GB)
 Version info: 'GraalVM dev Java 20.0.1+2-jvmci-23.0-b06 EE'
 Java version info: '20.0.1+2-jvmci-23.0-b06'
 C compiler: gcc (redhat, x86_64, 8.3.1)
 Garbage collector: Serial GC (max heap size: unlimited)
 5 user-specific feature(s)
 - com.oracle.svm.test.AbstractServiceLoaderTest$TestFeature
 - com.oracle.svm.test.NativeImageResourceUtils$TestFeature
 - com.oracle.svm.test.NoProviderConstructorServiceLoaderTest$TestFeature
 - com.oracle.svm.test.SerializationRegistrationTestFeature
 - com.oracle.svm.test.jfr.JfrTestFeature
# Printing compilation-target information to: /b/b/e/main/substratevm-enterprise/svmbuild/reports/target_info_20230209_103138.txt
Warning: Class java.lang.Number has subclasses. No classes were registered for object serialization.

# Printing native-library information to: /b/b/e/main/substratevm-enterprise/svmbuild/reports/native_library_info_20230209_103218.txt
[2/8] Performing analysis...  [****]                                                                    (41.5s @ 0.91GB)
   4,311 (78.35%) of  5,502 types reachable
   5,651 (53.41%) of 10,581 fields reachable
  22,691 (52.61%) of 43,134 methods reachable
   1,243 types,   211 fields, and 1,173 methods registered for reflection
      58 types,    56 fields, and    52 methods registered for JNI access
       4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (9.9s @ 1.38GB)
[4/8] Parsing methods...      [***]                                                                      (6.7s @ 3.65GB)
[5/8] Inlining methods...     [***]                                                                      (3.3s @ 0.73GB)
[6/8] Compiling methods...    [********]                                                                (75.2s @ 1.16GB)
[7/8] Layouting methods...    [****]                                                                    (20.6s @ 2.58GB)
[8/8] Creating image...                                                                                 (31.9s @ 3.33GB)
   9.79MB (17.76%) for code area:    12,014 compilation units
  10.20MB (18.50%) for image heap:  193,471 objects and 90 resources
  34.88MB (63.25%) for debug info generated in 14.8s
 270.86KB ( 0.48%) for other data
  55.14MB in total
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   6.56MB java.base                                            1.87MB byte[] for code metadata
   1.90MB svm.jar (Native Image)                               1.24MB byte[] for java.lang.String
 753.38KB jdk.jfr                                              1.21MB java.lang.String
 166.86KB com.oracle.svm.svm_enterprise                        1.04MB byte[] for general heap data
 142.04KB java.logging                                       719.80KB java.lang.Class
  57.97KB jdk.proxy1                                         380.47KB java.util.concurrent.ConcurrentHashMap$Node
  32.39KB org.graalvm.nativeimage.base                       337.13KB byte[] for embedded resources
  24.02KB jdk.proxy3                                         309.16KB java.util.HashMap$Node
  23.32KB org.graalvm.sdk                                    293.22KB java.lang.Object[]
  15.02KB jdk.internal.vm.compiler                           251.48KB byte[] for reflection metadata
  70.46KB for 12 more packages                                 2.49MB for 1187 more object types
------------------------------------------------------------------------------------------------------------------------
                        7.9s (3.8% of total time) in 64 GCs | Peak RSS: 6.85GB | CPU load: 7.40
------------------------------------------------------------------------------------------------------------------------
Produced artifacts:
 /b/b/e/main/substratevm-enterprise/svmbuild/hello.hello (executable, debug_info)
 /b/b/e/main/substratevm-enterprise/svmbuild/hello.hello.debug (debug_info)
 /b/b/e/main/substratevm-enterprise/svmbuild/sources (debug_info)
========================================================================================================================
Finished generating 'hello.hello' in 3m 25s.
GNU gdb (GDB) 10.2
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /b/b/e/main/substratevm-enterprise/svmbuild/hello.hello...
Reading symbols from /b/b/e/main/substratevm-enterprise/svmbuild/hello.hello.debug...
(gdb) set listsize 1
(gdb) show version
GNU gdb (GDB) 10.2
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word".

Found gdb version 10.2
Testing with isolates enabled!
Warning: cannot test printing of objects!
(gdb) set pagination off
(gdb) set print pretty on
(gdb) set print asm-demangle on
(gdb) set print symbol off
(gdb) ptype 'hello.Hello$DefaultGreeter'
type = class hello.Hello$DefaultGreeter : public hello.Hello$Greeter {
  public:
    void DefaultGreeter();
  private:
    void greet();
    int hashCode();
}

(gdb) break hello.Hello::main
Breakpoint 1 at 0x634470: file hello/Hello.java, line 215.

(gdb) run

Thread 1 "hello.hello" hit Breakpoint 1, hello.Hello::main(java.lang.String[]*) (args=0x7f491e701098) at hello/Hello.java:215
215	        Greeter greeter = Greeter.greeter(args);
(gdb) delete breakpoints
(gdb) list
215	        Greeter greeter = Greeter.greeter(args);

(gdb) backtrace
#0  hello.Hello::main(java.lang.String[]*) (args=0x7f491e701098) at hello/Hello.java:215
#1  0x000000000040fb35 in com.oracle.svm.core.JavaMainWrapper::runCore0() () at com/oracle/svm/core/JavaMainWrapper.java:177
#2  0x00000000004be605 in com.oracle.svm.core.JavaMainWrapper::runCore() () at com/oracle/svm/core/JavaMainWrapper.java:136
#3  com.oracle.svm.core.JavaMainWrapper::doRun(int, org.graalvm.nativeimage.c.type.CCharPointerPointer*) (argc=<optimized out>, argv=<optimized out>) at com/oracle/svm/core/JavaMainWrapper.java:232
#4  com.oracle.svm.core.JavaMainWrapper::run(int, org.graalvm.nativeimage.c.type.CCharPointerPointer*) (argc=<optimized out>, argv=<optimized out>) at com/oracle/svm/core/JavaMainWrapper.java:218
#5  com.oracle.svm.core.code.IsolateEnterStub::JavaMainWrapper_run_e6899342f5939c89e6e2f78e2c71f5f4926b786d(int, org.graalvm.nativeimage.c.type.CCharPointerPointer*) (__0=<optimized out>, __1=<optimized out>) at com/oracle/svm/core/code/IsolateEnterStub.java:1

(gdb) info args
(gdb) info locals
(gdb) info func greet
All functions matching regular expression "greet":

File hello/Hello.java:
37:	hello.Hello$Greeter *hello.Hello$Greeter::greeter(java.lang.String[]*);
71:	void hello.Hello$NamedGreeter::greet();

File hello/Target_hello_Hello_DefaultGreeter.java:
48:	void hello.Hello$DefaultGreeter::greet();

(gdb) step
hello.Hello$Greeter::greeter(java.lang.String[]*) (args=0x7f491e701098) at hello/Hello.java:37
37	            if (args.length == 0) {
(gdb) list
37	            if (args.length == 0) {

(gdb) ptype 'hello.Hello$NamedGreeter'
type = class hello.Hello$NamedGreeter : public hello.Hello$Greeter {
  private:
    _z_.java.lang.String *name;

  public:
    void NamedGreeter(java.lang.String *);
    void greet(void);
}

(gdb) ptype 'hello.Hello$Greeter'
type = class hello.Hello$Greeter : public java.lang.Object {
  public:
    void Greeter(void);
    static hello.Hello$Greeter * greeter(java.lang.String[] *);
}

(gdb) ptype 'java.lang.Object'
type = class java.lang.Object : public _objhdr {
  public:
    void Object(void);
    boolean equals(java.lang.Object *);
  private:
    java.lang.Class * getClass(void);
    int hashCode(void);
    void notify(void);
    void notifyAll(void);
  public:
    java.lang.String * toString(void);
    void wait(void);
  private:
    void wait(long);
  public:
    void wait(long, int);
}

(gdb) ptype _objhdr
type = struct _objhdr {
    _z_.java.lang.Class *hub;
    int idHash;
}

(gdb) ptype 'java.lang.String[]'
type = class java.lang.String[] : public java.lang.Object {
  public:
    int len;
    _z_.java.lang.String *data[0];
}

(gdb) backtrace
#0  hello.Hello$Greeter::greeter(java.lang.String[]*) (args=0x7f491e701098) at hello/Hello.java:37
#1  0x0000000000634489 in hello.Hello::main(java.lang.String[]*) (args=<optimized out>) at hello/Hello.java:215
#2  0x000000000040fb35 in com.oracle.svm.core.JavaMainWrapper::runCore0() () at com/oracle/svm/core/JavaMainWrapper.java:177
#3  0x00000000004be605 in com.oracle.svm.core.JavaMainWrapper::runCore() () at com/oracle/svm/core/JavaMainWrapper.java:136
#4  com.oracle.svm.core.JavaMainWrapper::doRun(int, org.graalvm.nativeimage.c.type.CCharPointerPointer*) (argc=<optimized out>, argv=<optimized out>) at com/oracle/svm/core/JavaMainWrapper.java:232
#5  com.oracle.svm.core.JavaMainWrapper::run(int, org.graalvm.nativeimage.c.type.CCharPointerPointer*) (argc=<optimized out>, argv=<optimized out>) at com/oracle/svm/core/JavaMainWrapper.java:218
#6  com.oracle.svm.core.code.IsolateEnterStub::JavaMainWrapper_run_e6899342f5939c89e6e2f78e2c71f5f4926b786d(int, org.graalvm.nativeimage.c.type.CCharPointerPointer*) (__0=<optimized out>, __1=<optimized out>) at com/oracle/svm/core/code/IsolateEnterStub.java:1

(gdb) next
38	                return new DefaultGreeter();
(gdb) backtrace 1
#0  hello.Hello$Greeter::greeter(java.lang.String[]*) (args=<optimized out>) at hello/Hello.java:38

(gdb) break hello.Hello$DefaultGreeter::greet
Breakpoint 2 at 0x634160: file hello/Target_hello_Hello_DefaultGreeter.java, line 48.

(gdb) delete breakpoints
(gdb) finish
0x0000000000634489 in hello.Hello::main(java.lang.String[]*) (args=<optimized out>) at hello/Hello.java:215
215	        Greeter greeter = Greeter.greeter(args);
Value returned is $1 = (hello.Hello$Greeter *) 0x7f491e701a90
(gdb) step
216	        greeter.greet();
(gdb) info args
(gdb) info locals
(gdb) break java.io.PrintStream::println
Breakpoint 3 at 0x4060af: java.io.PrintStream::println. (92 locations)

(gdb) continue

Thread 1 "hello.hello" hit Breakpoint 3, java.io.PrintStream::println(java.lang.String*) (this=0x7f491e1f43f8, x=0x7f491dea0ec8) at java/io/PrintStream.java:1167
1167	        if (getClass() == PrintStream.class) {
(gdb) info args
this = 0x7f491e1f43f8
x = 0x7f491dea0ec8

(gdb) ptype this
type = class java.io.PrintStream : public java.io.FilterOutputStream {
  private:
    boolean autoFlush;
    boolean closing;
    _z_.jdk.internal.misc.InternalLock *lock;
    _z_.java.nio.charset.Charset *charset;
    _z_.java.io.BufferedWriter *textOut;
    _z_.java.io.OutputStreamWriter *charOut;

  public:
    void PrintStream(java.io.OutputStream *);
  private:
    void PrintStream(boolean, java.io.OutputStream *);
  public:
    void PrintStream(java.io.OutputStream *, boolean);
    java.nio.charset.Charset * charset(void);
    void close(void);
  private:
    void ensureOpen(void);
  public:
    void flush(void);
  private:
    void implClose(void);
    void implFlush(void);
    void implNewLine(void);
    void implWrite(java.lang.String *);
    void implWrite(byte [] *, int, int);
    void implWriteln(java.lang.String *);
    void newLine(void);
  public:
    void print(java.lang.String *);
    void println(void);
    void println(java.lang.Object *);
    void println(java.lang.String *);
  private:
    static java.lang.Object * requireNonNull(java.lang.Object *, java.lang.String *);
    void write(java.lang.String *);
  public:
    void write(byte [] *, int, int);
  private:
    void writeln(java.lang.String *);
} *

(gdb) ptype 'hello.Hello'
type = class hello.Hello : public java.lang.Object {
  public:
    static _z_.java.util.function.Supplier *lambda;

    static void <clinit>(void);
  private:
    static void inlineA(void);
    static void inlineCallChain(void);
    static void inlineFrom(void);
    static void inlineHere(int);
    static void inlineIs(void);
    static void inlineMee(void);
    static void inlineMixTo(int);
    static void inlineMoo(void);
    static void inlineReceiveConstants(byte, int, long, java.lang.String *, float, double);
    static void inlineTailRecursion(int);
    static void inlineTo(int);
    static java.lang.String * lambda$static$0(void);
  public:
    static void main(java.lang.String[] *);
  private:
    static void noInlineFoo(void);
    static void noInlineHere(int);
    static void noInlineManyArgs(int, int, int, int, boolean, int, int, long, int, long, float, float, float, float, double, float, float, float, float, double, boolean, float);
    static void noInlinePassConstants(void);
    static void noInlineTest(void);
    static void noInlineThis(void);
}

(gdb) info func nline
All functions matching regular expression "nline":

File com/oracle/svm/core/genscavenge/GreyToBlackObjRefVisitor.java:
69:	boolean com.oracle.svm.core.genscavenge.GreyToBlackObjRefVisitor::visitObjectReferenceInline(org.graalvm.word.Pointer*, int, bool, java.lang.Object*);

File com/oracle/svm/core/thread/Safepoint.java:
335:	void com.oracle.svm.core.thread.Safepoint::notInlinedLockNoTransition();

File com/sun/proxy/jdk/proxy3/Proxy9.java:
	boolean com.sun.proxy.jdk.proxy3.$Proxy9::mayBeInlined();

File hello/Hello.java:
77:	void hello.Hello::noInlineFoo();
126:	void hello.Hello::noInlineHere(int);
162:	void hello.Hello::noInlineManyArgs(int, int, int, int, bool, int, int, long, int, long, float, float, float, float, double, float, float, float, float, double, bool, float);
188:	void hello.Hello::noInlinePassConstants();
112:	void hello.Hello::noInlineTest();
97:	void hello.Hello::noInlineThis();

Non-debugging symbols:
0x00007f511dc795b0  xdrrec_inline
0x00007f511dc86180  xdrmem_inline
0x00007f511dc86580  x_inline
0x00007f511dc866c0  xdrstdio_inline

(gdb) list inlineIs
102	        inlineA();

(gdb) list inlineA
107	        noInlineTest();

(gdb) delete breakpoints
(gdb) break hello.Hello::inlineIs
Checker break inlineIs: match 0 failed at line 0 Breakpoint 4 at 0x6368c4: file hello/Hello.java, line 102.

Checker break inlineIs {
  re.compile('Breakpoint [0-9]+ at 0x[0-9a-f]+: (hello\\.Hello::inlineIs\\. \\(2 locations\\)|file hello/Hello\\.java, line 128\\.)')
}

Breakpoint 4 at 0x6368c4: file hello/Hello.java, line 102.

WARNING: psutil is not available, java process detection is less accurate
  File "/b/b/e/mxtool/mx_enter.py", line 44, in <module>
    mx._main_wrapper()
  File "/b/b/e/mxtool/mx.py", line 18385, in _main_wrapper
    main()
  File "/b/b/e/mxtool/mx.py", line 18366, in main
    retcode = c(command_args)
  File "/b/b/e/mxtool/mx_commands.py", line 145, in __call__
    return self.command_function(*args, **kwargs)
  File "/b/b/e/mxtool/mx_gate.py", line 450, in gate
    _run_gate(cleanArgs, args, tasks)
  File "/b/b/e/mxtool/mx_gate.py", line 697, in _run_gate
    runner(args, tasks)
  File "/b/b/e/main/substratevm-enterprise/mx.substratevm-enterprise/mx_substratevm_enterprise.py", line 227, in svm_gate_body
    _gate_run_svmtests(tasks, args)
  File "/b/b/e/main/substratevm-enterprise/mx.substratevm-enterprise/mx_substratevm_enterprise.py", line 432, in _gate_run_svmtests
    debuginfotest(base_args + ['-H:+UseSerialGC', '-H:+UseCompressedReferences'])
  File "/b/b/e/mxtool/mx_commands.py", line 145, in __call__
    return self.command_function(*args, **kwargs)
  File "/b/b/e/graal/substratevm/mx.substratevm/mx_substratevm.py", line 1267, in debuginfotest
    native_image_context_run(
  File "/b/b/e/graal/substratevm/mx.substratevm/mx_substratevm.py", line 900, in native_image_context_run
    func(native_image, func_args)
  File "/b/b/e/graal/substratevm/mx.substratevm/mx_substratevm.py", line 1269, in <lambda>
    _debuginfotest(native_image, output_path, build_only, with_isolates_only, a), unmask(parsed.image_args),
  File "/b/b/e/graal/substratevm/mx.substratevm/mx_substratevm.py", line 864, in _debuginfotest
    mx.run([os.environ.get('GDB_BIN', 'gdb'), '-ex', 'python "ISOLATES=True"', '-x', testhello_py, hello_binary])
  File "/b/b/e/mxtool/mx.py", line 13628, in run
    abort(retcode)
  File "/b/b/e/mxtool/mx.py", line 4211, in abort
    traceback.print_stack()

Traceback (most recent call last):
gate: 09 Feb 2023 10:35:00(+06:54) END:   debuginfotest [0:03:50.630348] [disk (free/total): 372.6GB/511.3GB]
gate: 09 Feb 2023 10:35:00(+06:54) ABORT: Gate [0:06:54.668081] [disk (free/total): 372.6GB/511.3GB]
  File "/b/b/e/mxtool/mx_gate.py", line 450, in gate
    _run_gate(cleanArgs, args, tasks)
  File "/b/b/e/mxtool/mx_gate.py", line 697, in _run_gate
    runner(args, tasks)
  File "/b/b/e/main/substratevm-enterprise/mx.substratevm-enterprise/mx_substratevm_enterprise.py", line 227, in svm_gate_body
    _gate_run_svmtests(tasks, args)
  File "/b/b/e/main/substratevm-enterprise/mx.substratevm-enterprise/mx_substratevm_enterprise.py", line 432, in _gate_run_svmtests
    debuginfotest(base_args + ['-H:+UseSerialGC', '-H:+UseCompressedReferences'])
  File "/b/b/e/mxtool/mx_commands.py", line 145, in __call__
    return self.command_function(*args, **kwargs)
  File "/b/b/e/graal/substratevm/mx.substratevm/mx_substratevm.py", line 1267, in debuginfotest
    native_image_context_run(
  File "/b/b/e/graal/substratevm/mx.substratevm/mx_substratevm.py", line 900, in native_image_context_run
    func(native_image, func_args)
  File "/b/b/e/graal/substratevm/mx.substratevm/mx_substratevm.py", line 1269, in <lambda>
    _debuginfotest(native_image, output_path, build_only, with_isolates_only, a), unmask(parsed.image_args),
  File "/b/b/e/graal/substratevm/mx.substratevm/mx_substratevm.py", line 864, in _debuginfotest
    mx.run([os.environ.get('GDB_BIN', 'gdb'), '-ex', 'python "ISOLATES=True"', '-x', testhello_py, hello_binary])
  File "/b/b/e/mxtool/mx.py", line 13628, in run
    abort(retcode)
  File "/b/b/e/mxtool/mx.py", line 4237, in abort
    raise SystemExit(error_code)
SystemExit: 1

The sequence of mx commands that were executed until the failure follows:

mx --kill-with-sigquit --strict-compliance --dy substratevm-enterprise-gcs version --oneline
mx --kill-with-sigquit --strict-compliance --dy substratevm-enterprise-gcs sversions
mx --kill-with-sigquit --strict-compliance --dy substratevm-enterprise-gcs verifymultireleaseprojects
mx --kill-with-sigquit --strict-compliance --dy substratevm-enterprise-gcs clean --all
mx --kill-with-sigquit --strict-compliance --dy substratevm-enterprise-gcs build -p --warning-as-error --force-javac
mx --kill-with-sigquit --strict-compliance --dy substratevm-enterprise-gcs debuginfotest --output-path /b/b/e/main/substratevm-enterprise/svmbuild -H:+UseSerialGC -H:+UseCompressedReferences

If the previous sequence is incomplete or some commands were executed programmatically use:

mx --kill-with-sigquit --strict-compliance --dy substratevm-enterprise-gcs gate --strict-mode --tags build,debuginfotest

WARNING: psutil is not available, java process detection is less accurate
  File "/b/b/e/mxtool/mx_enter.py", line 44, in <module>
    mx._main_wrapper()
  File "/b/b/e/mxtool/mx.py", line 18385, in _main_wrapper
    main()
  File "/b/b/e/mxtool/mx.py", line 18366, in main
    retcode = c(command_args)
  File "/b/b/e/mxtool/mx_commands.py", line 145, in __call__
    return self.command_function(*args, **kwargs)
  File "/b/b/e/mxtool/mx_gate.py", line 457, in gate
    total.abort(str(e))
  File "/b/b/e/mxtool/mx_gate.py", line 212, in abort
    mx.abort(codeOrMessage)
  File "/b/b/e/mxtool/mx.py", line 4211, in abort
    traceback.print_stack()
1
2023-02-09 10:35:00(+07:15)[ERROR] Process failed with return code 1
=======================================================================================
```
</details>

Similar failure on `gate-svm-enterprise-debuginfotest-jdk20-linux-aarch64`

**Review by fniephaus — APPROVED** [review]

**Comment by fniephaus** [comment]
Our CI has unfortunately mirrored this to https://github.com/oracle/graal/pull/6895.

File: substratevm/mx.substratevm/testhello.py
Comment (fniephaus) [inline substratevm/mx.substratevm/testhello.py:461]: Why is this commented out?

Comment (zakkak) [reply substratevm/mx.substratevm/testhello.py:461]: Accidentally, good catch! Do you want me to fix it here?

Comment (fniephaus) [reply substratevm/mx.substratevm/testhello.py:461]: Yes please. :)

Comment (zakkak) [reply substratevm/mx.substratevm/testhello.py]: done

**Pull request merged**
